src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -94,9 +94,14 @@ public static void ThrowJsonException(string? message = null)
 
         [DoesNotReturn]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        public static void ThrowInvalidOperationException_CannotSerializeOpenGeneric(Type type)
+        public static void ThrowInvalidOperationException_CannotSerializeInvalidType(Type type, Type? parentClassType, PropertyInfo? propertyInfo)
```

Comment:
Why is `PropertyInfo` nullable here?

Reply:
`parentClassType` & `propertyInfo` will both be `null` when an invalid type is detected during the creation of a `JsonClassInfo` info for the type.

If an invalid type is detected as the return type of a property on a POCO (when creating `JsonPropertyInfo`s), then `parentClassType` and `propertyInfo` will both not be `null`.

I added an assert that `propertyInfo` is `null` when `parentClassInfo` is `null`.

Not sure adding a `Debug.Assert` right before the last throw adds much value, as there'll be a null ref anyway (which will be encountered in debug mode) if `propertyInfo` is null.